clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2726,6 +2726,35 @@ namespace {
      }
    }
 
+   // When a variable goes out of scope:
+   // 1) it has to be removed from ObservedBounds in the CheckingState
+   //    if it is a checked pointer variable because we no longer want
+   //    to validate its bounds, and
+   // 2) the expressions in EquivExprs that use it have to be removed
+   //    because the expressions are now undefined.
+   void UpdateStateForVariableOutOfScope(CheckingState &State, VarDecl *V) {
+     if (V->hasBoundsExpr()) {
+       const AbstractSet *A = AbstractSetMgr.GetOrCreateAbstractSet(V);
+       auto I = State.ObservedBounds.find(A);
+       if (I != State.ObservedBounds.end())
+         State.ObservedBounds.erase(A);
+     }
+
+     EquivExprSets CrntEquivExprs(State.EquivExprs);
+     State.EquivExprs.clear();
+     for (auto I = CrntEquivExprs.begin(); I != CrntEquivExprs.end(); ++I) {
```

Comment:
Since `CrntEquivExprs` is not mutated inside the loop, a more efficient way to write this loop is as follows:
`for (auto I = CrntEquivExprs.begin(), E = CrntEquivExprs.end(); I != E; ++I)`

This avoids repeated evaluation of `CrntEquivExprs.end()` each time through the loop.

See https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop

Reply:
Done.